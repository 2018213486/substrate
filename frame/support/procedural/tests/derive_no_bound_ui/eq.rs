trait Trait {
	type C;
}

#[derive(
	frame_support_procedural::EqNoBound,
)]
struct Foo<T: Trait> {
	c: T::C,
}

fn main() {}
