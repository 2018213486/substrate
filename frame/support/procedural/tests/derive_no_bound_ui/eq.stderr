error[E0277]: can't compare `Foo<T>` with `Foo<T>`
   --> $DIR/eq.rs:8:8
    |
8   | struct Foo<T: Trait> {
    |        ^^^ no implementation for `Foo<T> == Foo<T>`
    |
   ::: $RUST/core/src/cmp.rs
    |
    | pub trait Eq: PartialEq<Self> {
    |               --------------- required by this bound in `std::cmp::Eq`
    |
    = help: the trait `std::cmp::PartialEq` is not implemented for `Foo<T>`
